src/components/codemirror.tsx

```diff
@@ -52,6 +52,14 @@ export default function CodeMirror(props: CodeMirrorProps) {
 		});
 	};
 
+	useEffect(() => {
+		if(props.roomState){
```

Comment:
it could be `isNewSaveStrat.value = props.roomState ? true : false`. it can't be `isNewSaveStrat.value = props.roomState`, bc they're not the same type. but u're right, will change that